tests/foreman/cli/test_hostgroup.py

```diff
@@ -43,245 +43,278 @@
-            Puppet.info({'name': mod['name'], 'puppet-environment': cls.env['name']})
-            for mod in puppet_modules
-        ]
-        cls.content_source = Proxy.list(
-            {'search': f'url = https://{settings.server.hostname}:9090'}
-        )[0]
-        cls.hostgroup = make_hostgroup(
-            {'content-source-id': cls.content_source['id'], 'organization-ids': cls.org['id']}
-        )
 
-    @pytest.mark.tier2
-    def test_negative_create_with_name(self):
-        """Don't create an HostGroup with invalid data.
-
-        :id: 853a6d43-129a-497b-94f0-08dc622862f8
-
-        :expectedresults: HostGroup is not created.
-        """
-        for name in invalid_values_list():
-            with self.subTest(name):
-                with self.assertRaises(CLIReturnCodeError):
-                    HostGroup.create({'name': name})
-
-    @pytest.mark.tier1
-    @pytest.mark.upgrade
-    def test_positive_create_with_multiple_entities_and_delete(self):
-        """Check if hostgroup with multiple options can be created and deleted
-
-        :id: a3ef4f0e-971d-4307-8d0a-35103dff6586
-
-        :expectedresults: Hostgroup should be created, has all defined
-            entities assigned and deleted
-
-        :BZ: 1395254, 1313056
-
-        :CaseLevel: Integration
-
-        :CaseImportance: Critical
-        """
-        # Common entities
-        name = valid_hostgroups_list()[0]
-        loc = make_location()
-        org = make_org()
-        orgs = [org, self.org]
-        env = make_environment({'location-ids': loc['id'], 'organization-ids': org['id']})
-        lce = make_lifecycle_environment({'organization-id': org['id']})
-        # Content View should be promoted to be used with LC Env
-        cv = make_content_view({'organization-id': org['id']})
-        ContentView.publish({'id': cv['id']})
-        cv = ContentView.info({'id': cv['id']})
-        ContentView.version_promote(
-            {'id': cv['versions'][0]['id'], 'to-lifecycle-environment-id': lce['id']}
-        )
-        # Network
-        domain = make_domain({'location-ids': loc['id'], 'organization-ids': org['id']})
-        subnet = make_subnet({'domain-ids': domain['id'], 'organization-ids': org['id']})
-        # Operating System
-        arch = make_architecture()
-        ptable = make_partition_table({'location-ids': loc['id'], 'organization-ids': org['id']})
-        os = make_os({'architecture-ids': arch['id'], 'partition-table-ids': ptable['id']})
-        os_full_name = "{} {}.{}".format(os['name'], os['major-version'], os['minor-version'])
-        media = make_medium(
+
+pytestmark = [
+    pytest.mark.skipif((not settings.repos_hosting_url), reason='Missing repos_hosting_url')
+]
+PUPPET_MODULES = [
+    {'author': 'robottelo', 'name': 'generic_1'},
+    {'author': 'robottelo', 'name': 'generic_2'},
+]
+
+
+@pytest.fixture(scope='module')
+def org():
+    """Create an organization."""
+    return make_org()
```

Comment:
This make_org() is from robottelo.cli.factory. Try to use API for setup when possible as its faster. See if you can share API fixtures between API and CLI module in the `pytest_fixtures/api_fixtures.py` file.

Reply:
Removed `org` fixture and used existing `module_org` fixture, and created `org_2` using nailgun as well.

Reply:
thanks